### ISSUE TITLE
Ruby 2.0.0-p353 (x64) fail to start on Windows 2003 server

### ISSUE THREAD

**Issue by lionelperrin** [oneclick/rubyinstaller#211]
Hello,

I tried to run ruby 64bits on a windows 2003 server. I tried the following links: http://dl.bintray.com/oneclick/rubyinstaller/rubyinstaller-2.0.0-p353-x64.exe and http://dl.bintray.com/oneclick/rubyinstaller/ruby-2.0.0-p353-x64-mingw32.7z. In both cases, ruby fail to start with the following error message:

"The procedure entry point _gmtime64_s could not be located in the dynamic link library msvcrt.dll." 

Actually, the msvcrt.dll on windows 2003 does not define this external symbol (There is no symbol xxxx_s in this version of the dll.)

As the ruby executable is built with mingw32, (and not visual studio), it looks that we can't solve this issue with the install of a Visual C++ Redistributable Package.

Is there any workaround ? Is there a limitation on using ruby-x64 ? Is it necessary to link with msvcrt.dll.

For info, ruby 32 bit works with this windows version.

Regards,

Lionel

**Comment by graf0** [comment]
Hi!

It seems that guys at puppet found solution: https://tickets.puppetlabs.com/browse/PUP-3780 (check last comment).

**Comment by ferventcoder** [comment]
Yes we did. :)

It's still only a solution for x86 binaries though

**Comment by Azolo** [comment]
Yeah, I've talked to the Puppet guys about what they're doing and I've talked to Luis about what it might take to change the compiler toolchain (aka the DevKit).

There are some problems with GDBM with the toolchain they're using (which is a blocker).
I want to upgrade the DevKit, but I want to make distribution easier by gemifying the DevKit.
The DevKit is already confusing for new people. So simplifying that would be ideal before adding another DevKit to choose from.

It's something I'm working on in my spare time.
